draft-ietf-avtcore-rtp-over-quic.md

```diff
@@ -536,19 +536,34 @@ different flow identifiers. RTP and RTCP packets of a single RTP session can use
 they can use different flow identifiers.
 
-The association between flow identifiers and RTP streams MUST be negotiated
-using appropriate signaling. The signaling happens out of band and thus a stream
-or DATAGRAM with a given flow identifer can arrive before the signaling
-finished. In that case, an endpoint cannot associate the stream or DATAGRAM with
-the corresponding RTP stream. The endpoint can buffer streams and DATAGRAMs
-using an unknown flow identifier until they can be associated with the
-corresponding RTP stream. To avoid resource exhaustion, the buffering endpoint
-MUST limit the number of streams and DATAGRAMs to buffer. If the number of
-buffered streams exceeds the limit on buffered streams, the endpoint MUST send a
-STOP_SENDING with the error code ROQ_UNKNOWN_FLOW_ID. It is an implementation's
-choice on which stream to send STOP_SENDING. If the number of buffered DATAGRAMs
-exceeds the limit on buffered DATAGRAMs, the endpoint MUST drop a DATAGRAMs. It
-is an implementation's choice which DATAGRAMs to drop.
+Endpoints need to associate flow identifiers with RTP streams. Depending on the
+context of the application, the association can be statically configured,
+signaled using an out-of-band signaling mechanism (e.g., SDP), or applications
+might be able to identify the stream based on the RTP packets sent on the stream
+(e.g., by inspecting the payload type).
+
+If an endpoint receives a flow identifier that it cannot associate with an RTP
+stream, the endpoint MAY close the connection using the ROQ_UNKNOWN_FLOW_ID
+error code. Closing the connection can be a valid response if it is not expected
+that out of band signaling is still ongoing and the application cannot handle
+unknown flow identifiers.
+
+If the association of flow identifiers with RTP streams depends on out-of-band
+signaling, the signaling mechanism SHOULD be completed before the exchange of
+RTP packets using the new flow identifiers starts.
+
+In cases where it cannot be guaranteed that signaling is completed before RTP
+packets are transmitted, streams or DATAGRAMs with a given flow identifer can
+arrive before the signaling finished. In that case, an endpoint cannot associate
+the stream or DATAGRAM with the corresponding RTP stream. The endpoint can
+buffer streams and DATAGRAMs using an unknown flow identifier until they can be
+associated with the corresponding RTP stream. To avoid resource exhaustion, the
+buffering endpoint MUST limit the number of streams and DATAGRAMs to buffer. If
+the number of buffered streams exceeds the limit on buffered streams, the
+endpoint MUST send a STOP_SENDING with the error code ROQ_UNKNOWN_FLOW_ID. It is
+an implementation's choice on which stream to send STOP_SENDING. If the number
+of buffered DATAGRAMs exceeds the limit on buffered DATAGRAMs, the endpoint MUST
+drop a DATAGRAMs. It is an implementation's choice which DATAGRAMs to drop.
```

Comment:
Uppercase DATAGRAM is defined in the terminology section as referring to a QUIC DATAGRAM frame because we tried to avoid confusion between QUIC *frames* and media *frames*.

Reply:
Yeah, and it's actually worse than what @mengelbart said. From [the current draft](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#name-terminology-and-notation):

>Datagram:
The term "datagram" is ambiguous. Without a qualifier, "datagram" could refer to a UDP packet, **or** a QUIC DATAGRAM frame, as defined in QUIC's unreliable DATAGRAM extension [[RFC9221](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#RFC9221)], **or** an RTP packet encapsulated in UDP, **or** an RTP packet capsulated in QUIC DATAGRAM frame. This document uses the uppercase "DATAGRAM" to refer to a QUIC DATAGRAM frame and the term RoQ datagram as a short form of "RTP packet encapsulated in a QUIC DATAGRAM frame".

We might consider adding an issue for "things that have surprised readers in the past". 

- There are some terms that we use, that really do require the reader to look them up in the terminology section. This is one of them. And I note that unqualified "frame" doesn't appear in the terminology section - maybe it should.